.github/workflows/ci.yaml

```diff
@@ -0,0 +1,83 @@
+name: CI
+on:
+  push: 
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  interface:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    defaults:
+      run:
+        working-directory: ./interface
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+
+      - name: Setup Node.js
+        uses: actions/setup-node@master
+        with:
+          node-version: '${{ steps.nvm.outputs.NVMRC }}'
+
+      - name: Install dependencies
+        run: yarn install --nonInteractive --frozen-lockfile --prefer-offline
+
+      - name: Build
+        run: yarn build
+
+  getImplementations:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ env.matrix }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - id: set-matrix
+        run: echo "matrix=$(./scripts/getImplementations.sh)" >> $GITHUB_ENV
+
+  implementations-ci:
+    runs-on: ubuntu-latest
+    needs:
+      - interface
+      - getImplementations
+    strategy:
+      matrix: ${{fromJSON(needs.getImplementations.outputs.matrix)}}
+    defaults:
+      run:
+        working-directory: ./implementations/${{ matrix.package }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+        working-directory: ./interface
+
+      - name: Setup Node.js
+        uses: actions/setup-node@master
+        with:
+          node-version: '${{ steps.nvm.outputs.NVMRC }}'
+
+      - name: Install dependencies
+        run: yarn install --nonInteractive --frozen-lockfile --prefer-offline
+        working-directory: ./interface
+
+      - name: Build
+        run: yarn build
+        working-directory: ./interface
```

Comment:
I think we can also simplify this by changing yarn build in every impl scripts:
```
scripts:
  build: yarn build:interface && yarn build:implementation
  build:interface: yarn --cwd <interface_dir>
  build:implementation: polywrap build
```
We can do above changes in package.json in every impls

Reply:
You said earlier that we don't need to install dependencies for the interface in each implementation job, but here you're saying we still need to build it each time. Could you clarify?

Reply:
Let's do this in a future PR

Reply:
Actually, after sleeping on it, I don't think this is a good idea. It makes the CI/CD depend on yarn scripts being a particular way. Having multiple steps in the CI/CD is more verbose, but that part of the CI/CD is simple and easy to understand.